### PR TITLE
[NFC] Fix c++ style comment in c file

### PR DESCRIPTION
**Pull request by jsji**
This is one of the many PRs to fix errors with LLVM_ENABLE_WERROR=on. Built by GCC 11.

Fix warnings:
llvm-project/clang/include/clang-c/Index.h:2983:3: error: C++ style comments are not allowed in ISO C90 [-Werror]
 2983 |   // HLSL Types

### CODE REVIEW

File: clang/tools/c-index-test/c-index-test.c [hunk 1]
```diff
@@ -8,7 +8,6 @@
 #include "clang-c/Documentation.h"
 #include "clang-c/Index.h"
 #include "clang/Config/config.h"
-#include "llvm/Support/AutoConvert.h"
```
Comment (abhina-sree) [inline clang/tools/c-index-test/c-index-test.c:11]: We need this header because we are using the function enablezOSAutoConversion below in this file

Comment (jsji) [reply clang/tools/c-index-test/c-index-test.c:11]: This is a c++ header, if we really need it, then we need to update it to c. @abhina-sree Can you please help to do so? Thanks.

Comment (abhina-sree) [reply clang/tools/c-index-test/c-index-test.c:11]: Inside that header we guard the C++ code inside an `#ifdef __cplusplus` so it should work for both C and C++. Are you hitting any errors as a result of this header file? it is also guarded with `__MVS__` so no other platforms outside of z/OS should be using these functions

Comment (jsji) [reply clang/tools/c-index-test/c-index-test.c:11]: Please update all the comments to use c style instead then?

Comment (abhina-sree) [reply clang/tools/c-index-test/c-index-test.c:11]: Yes, I think that would be a better solution, I will try to get to that when I have time, thanks!

Comment (jsji) [reply clang/tools/c-index-test/c-index-test.c]: Done. Please review. Thanks.

**Review by abhina-sree — APPROVED** [review]
LGTM, thanks!

**Pull request merged**